### DISCUSSION TITLE
【1.0.0 ドックフーディング】不明点・修正点まとめ

### DISCUSSION THREAD

**Discussion by pss-aileen** [first-contributions-ja/first-contributions-ja.github.io#150] [【運営】相談・報告]
Discordの件の、不明点・修正検討の部分はこちらで！
- バージョン1のリリースに含めるものは基本ドキュメント関連
- コード、サイトの裏側は1.0.0以降で

---
## 確認用トピックリスト
（以下のコメントのトピックをリストにして、それぞれissue化して紐付けたい。
ほぼ確定したことと、議論が必要な問題の整理と、イシュー化を忘れないために。）

例）
- Step4:
  - [ ] 確定）`,`が不要なので修正する→ #イシューナンバー
  - [ ] 確定）カラーコード、絵文字を参照できるサイトのリンクを記載する

**Comment by pss-aileen** [comment]
### Step 7:プルリクエスト（PR）を作成 でどこのブランチに向けてマージするか

- マージする際ってmainに確か向かった気がする
- 一応 main に向けてマージと書いておいたほうがよさそう

**Reply by kagomen** [reply]
```
https://github.com/ユーザーネーム/first-contributions-ja
```
↓
```
https://github.com/first-contributions-ja/first-contributions-ja.github.io
```

ですかね？
`https://github.com/kagomen/first-contributions-ja.github.io`に飛んだらプルリクできない謎のページ？に行き着きました

**Reply by pss-aileen** [reply]
![スクリーンショット 2024-04-01 1 07 11](https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/140990100/373a4406-2a65-4676-bd33-10948dfb4e74)

ここの場面ですね、大丈夫かと思いますが、mainに向かっていることを確認するようにしてもよいのかなと

**Reply by pss-aileen** [reply]
https://github.com/first-contributions-ja/first-contributions-ja.github.io/discussions/150#discussioncomment-8971006

**Comment by kazzyfrog** [comment]
## Step2:
順番を変更した方がいいかも
現在
- ターミナルを開き、リポジトリをクローンするディレクトリに移動します。
- GitHub 上の、フォークしたリポジトリのページに移動し、緑色の "Code" ボタンからクローンする URL をコピーします。
- 以下のコマンドを使用して、フォークしたリポジトリをローカルマシンにクローンします：
↓
- GitHub 上の、フォークしたリポジトリのページに移動し、緑色の "Code" ボタンからクローンする URL をコピーします。
- ターミナルを開き、リポジトリをクローンするディレクトリに移動します。
- 以下のコマンドを使用して、フォークしたリポジトリをローカルマシンにクローンします：

**Comment by kazzyfrog** [comment]
## STEP3

- 次のコマンドでリポジトリのディレクトリ内に移動します：cd first-contributions-ja
これはミス
正しくは、
- 次のコマンドでリポジトリのディレクトリ内に移動します：cd first-contributions-ja.github.io

**Reply by pss-aileen** [reply]
こちらは修正確定かと思います

**Comment by kazzyfrog** [comment]
## Step4

`  "favoriteEmoji": "あなたの好きな絵文字",`
これの末尾のカンマ`,`が不要。

**Reply by kazzyfrog** [reply]
あなたの好きなカラーコード
に関しては、カラーコードを見れるツールのリンク貼った方が丁寧かも！

**Reply by kagomen** [reply]
- やっぱり絵文字3つくらい登録したい！
- ↑のカラーコードと同じように、絵文字に関しても[絵文字リスト](https://gist.github.com/rxaviers/7360908)を載せてもいいかも
- 変更を保存するのを忘れないで！みたいな文言ほしいかも

**Reply by pss-aileen** [reply]
- `,` 不要同意
- カラーコード 同意 日本カラー良き
  - https://www.colordic.org/y
  - https://nipponcolors.com/ ⭐️
  - https://yeun.github.io/open-color/
  - https://colorhunt.co/palettes/pastel
- 絵文字リストも同意！

**Comment by kazzyfrog** [comment]
その他思ったこと：

やっぱりJSONファイルの整形は必須かも。
コードを貼り付けて、インデントなどが揃っていないと、少し戸惑う。。

所感：
なんか、わかりにくい・迷ってしまう、、などと感じる部分は多々ありそう！と思ったが、
いかんせん自分で作ったから、違和感なくすんなり出来てしまった。
もっとフィードバックが欲しいところです！

**Reply by pss-aileen** [reply]
まだドックフーディングしてませんが
逆にインデント自分でしてもらう、調整してもらうというのもありかもしれません！

コーディングのルールに従おう的な感じで

**Comment by pss-aileen** [comment]
## マージ時のメッセージ


> プルリクエスト確認しました！
> LGTM✨

> 以上でコントリビューションは終了です！

> 参加してくれてありがとうございました！
> よろしければ参加のご感想をお聞かせください！

的なテンプレでもよいので
あると「これでちゃんとマージされたんだな！」やった！と思えそそう。

**Reply by kagomen** [reply]
<img width="942" alt="スクリーンショット 2024-03-31 15 09 43" src="https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/154225199/d45d5550-63a3-4a3c-bca1-b3d6e1de523a">

こちら、あった方が絶対良いなと思いました！✨
また、

> よろしければ参加のご感想を[Discussion](https://github.com/first-contributions-ja/first-contributions-ja.github.io/discussions/categories/-%E3%83%81%E3%83%A5%E3%83%BC%E3%83%88%E3%83%AA%E3%82%A2%E3%83%AB-%E6%84%9F%E6%83%B3)でお聞かせください！

のようにリンクを貼ったらよりスムーズだと思いました！

**Comment by kagomen** [comment]
## STEP前

### このプロジェクトの目的は？
- しずらい→しづらい

### どんな貢献をすればいいの？
- 貢献の方法はわかるけど、どういう結果になるのかがわからない（これは個別ページを用意してないからしょうがないですよね）
- 最後に「それでは詳しく説明していきます！実際に手を動かしてみてみましょう！」的な文言があると流れが自然だなと思った

**Reply by pss-aileen** [reply]
> 貢献の方法はわかるけど、どういう結果になるのかがわからない
- 「TOPページの人数変わるよ！」的な、個人の情報の入力は今後のアップデートでページやカードが表示できるようになる...と書いておくか...！
- イメージがあると良いですよね

**Comment by kagomen** [comment]
## STEP 1
フォークしたものの、自分が何をしたのかがよくわからなかった
- STEP2以降の作業とどう繋がってるのかがよくわからない（その解説までする必要はない気もするが・・・）

**Comment by kagomen** [comment]
## STEP 5
エディタからターミナルに作業場が移動してるので、「ターミナルに移動し、」みたいな文言があるとよりわかりやすいと思った

**Comment by kagomen** [comment]
## Finish
「それ」→「あなたの変更内容は」（個人的にちょっと気になった）

**Comment by kagomen** [comment]
## さらなるステップ

> もし興味があれば、[その他の貢献](https://github.com/first-contributions-ja/first-contributions-ja.github.io?tab=readme-ov-file#%E3%81%9D%E3%81%AE%E4%BB%96%E3%81%AE%E8%B2%A2%E7%8C%AE)を確認してください。
または、[Discussions](https://github.com/first-contributions-ja/first-contributions-ja.github.io/discussions)にて、このプロジェクトに関する感想を聞かせてください 🙌

- 順番逆でも良いかも？よりイージーなものを前にした方が良い気が？（感想書いてほしいですし！）

- 引用になってるところを、CONTRIBUTING.mdのように[Alert](
https://docs.github.com/ja/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts) で書いた方が良い気がした

> 完全日本語のチュートリアルで、OSS 活動を始める 🚀

READMEの一番最初のここも！

**Comment by pss-aileen** [comment]
## はじめに
- オープンソースがそもそもわからない人への一言解説があっても良いかと思った
- コラボレーションってなんだろうとやはり思う
- 概念の説明が欲しい

**Comment by pss-aileen** [comment]
## 参加方法
- 「どんな貢献をするか」ではなく、「何をするのか？」
  - 貢献だと、なんだか飲み込めない
- 記載する情報は以下の通り、のコードは `{}` があったほうが、現実のものに近いと思った

### どんな貢献をすればいいの？
 > このプロジェクトの参加者として、Contributors.json ファイルに必要な情報を入力し、プロフィールを作成します！
引用にするとちょっと存在感がうすくなってここわりと大切だけどスルーしちゃいそう

「今回の貢献はプロジェクトの参加者として、Contributors（貢献者一覧）ファイルにあなたの情報を入力しこのリポジトリにアップロード(？）します。」

**Comment by pss-aileen** [comment]
## 実際に手を動かす
- 8 つの簡単なステップで、オープンソースのコントリビューターになる 😎
  - これは引用ではなく、太字でもいいかも。

**Reply by pss-aileen** [reply]
他、まだありますが明日アップします。

**Comment by pss-aileen** [comment]
## Step1

どこまで親切にするかですが、こういう画面がでるよ〜と、
Copy the main branch onlyはチェックのままでOKだよ〜まで書くと親切かと思った

![ 4](https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/140990100/208df2de-8265-46e9-a9d5-d8e2a4af44a7)

**Comment by pss-aileen** [comment]
## Step2
- フォークとクローンの概念が欲しいなぁ〜と思った、日本語リポとして少し噛み砕ければなと
- フォーク [リポジトリをフォークする - GitHub Docs](https://docs.github.com/ja/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo#about-forks)
  - > フォークは、上流リポジトリに再提案する前に、アイデアや変更に繰り返し取り組むために使用されます。たとえば、オープンソース プロジェクトや、ユーザーが上流リポジトリへの書き込みアクセス権を持っていない場合などです。
- クローン [リポジトリをクローンする - GitHub Docs](https://docs.github.com/ja/repositories/creating-and-managing-repositories/cloning-a-repository#about-cloning-a-repository)
  - 全く同じもの、コピー
  - > マージの競合の修正、ファイルの追加や削除、より大きなコミットのプッシュを簡単に行うことができます。  

---

> 緑色の "Code" ボタンからクローンする URL をコピーします。

修正提案）緑色の "Code" ボタンをクリックし、HTTPS の URL ををコピーする。

![スクリーンショット 2024-04-01 17 03 56](https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/140990100/bc721bab-fbec-4068-9d74-1b46b7cbd182)

**Reply by pss-aileen** [reply]
そう、そもそもOSSコントリビュートはフォークするなんて知らなかった。
クローンしてそのまま参加できるものだと思っていた
権限周りのこと知らなかった、

**Comment by pss-aileen** [comment]
## Step3
- クローンした後、フォルダに移動してからはVSCodeのターミナルで実行するのはどうか？
- クローンして、フォルダができあるまでと、ブランチをわけてからの作業はわけたほうが作業の分離？というか、わかりやすい気がする

なので、個人的にやるとしたら
```
cd first-contributions-ja
```
として、
```
code .
```
と入力して、コードエディタを開きます。...
と、思ったのですが `code .` はVSCodeでフォルダを開くコマンドだったので保留

ただ、この段階でエディタを開く方がよいかな〜と
それか、エディタをあえて絞って作ってやってみるか

あ、あと、好きなコードエディタである場合だったとしても
最初に使うものはお好きなエディタとターミナル！とどこかで書いておくと良いかも。


---

### 修正提案1
#### 現状
次のコマンドでリポジトリのディレクトリ内に移動します：`cd first-contributions-ja`

#### 改善案
コピーできるようにコードブロックにする

- 次のコマンドでリポジトリのディレクトリ内に移動します
  ```shell
  cd first-contributions-ja
  ```
  
### 修正提案2
#### 現状
- プロジェクトの`main`ブランチからあなたの作業を切り離すために、新しいブランチを作成してください。
  - 以下のコマンドを使用できます：
  ```
  git switch -c <ブランチ名>
  ```
- ブランチ名は、あなたがこれから行う作業に関連した意味のある名前を付けます。
  - 例えば以下のようになります（`yourname`をあなたのハンドルネームに置き換えることを忘れないでください）：
  ```
  git switch -c add-yourname
  ```

#### 改善案

- プロジェクトのmainブランチから、あなたの作業を切り離すため、新ブランチを作成します。
- `yourname` には、あなたのハンドルネームに置き換えてください
- ひらがな、漢字、半角空白は使わず、半角英数字で入力してください
- ❌: おぎのちひろ、荻野ちひろ、Ogino Chihiro

```bash
git switch -c add-yourname
```

例）

```bash
git switch -c add-oginochihiro
```

**Comment by pss-aileen** [comment]
## Step4
Step3かStep4で
```
git status
```
を使って、きちんと新しいブランチにいるか確認できるようにすると良いかも

**Comment by pss-aileen** [comment]
## 全体 
`yourname` とされているところは、例として全部 oginochihiro 入れておいたら実際の入力のイメージわくかも
よくあるが「yourname」と書かれていてそこに入れるにも実際どういれたらいいのかの例を見たいと思うことがあるので...

**Comment by pss-aileen** [comment]
## Step7

プルリクエスト作成時に、どのリポジトリのどのブランチから、first-contributions-jsのmainに向かっているかを確認してもらいたい
一応。間違いないように。

あと、タイトル、概要欄はどうすればいいのか。
タイトルはそのまま、概要は空欄で良いのかなど

![スクリーンショット 2024-04-01 17 25 44](https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/140990100/c2a04806-1aa2-4103-a60d-179315d2a47d)

**Reply by pss-aileen** [reply]
あと、最終的にTOPに人数が表示されるけど、そこが変わっているかは確認させなくて良いか？
現状そこが連動しているので....
npm run devで...！

**Reply by pss-aileen** [reply]
ざっと以上かな、と思います！
